river/ensemble/adaptive_random_forest.py

```diff
@@ -0,0 +1,824 @@
+                 stop_mem_management: bool = False,
+                 remove_poor_attrs: bool = False,
+                 merit_preprune: bool = True,
+                 leaf_prediction: str = 'model',
+                 leaf_model: base.Regressor = None,
+                 model_selector_decay: float = 0.95,
+                 nominal_attributes: list = None,
+                 max_depth: int = None,
+                 seed=None):
+        super().__init__(
+            n_models=n_models,
+            max_features=max_features,
+            lambda_value=lambda_value,
+            metric=metric,
+            disable_weighted_vote=disable_weighted_vote,
+            drift_detector=drift_detector,
+            warning_detector=warning_detector,
+            seed=seed
+        )
+
+        self._n_samples_seen = 0
+        self._base_member_class = ForestMemberRegressor
+
+        # Tree parameters
+        self.max_size = max_size
+        self.memory_estimate_period = memory_estimate_period
+        self.grace_period = grace_period
+        self.split_confidence = split_confidence
+        self.tie_threshold = tie_threshold
+        self.binary_split = binary_split
+        self.stop_mem_management = stop_mem_management
+        self.remove_poor_attrs = remove_poor_attrs
+        self.merit_preprune = merit_preprune
+        self.leaf_prediction = leaf_prediction
+        self.leaf_model = leaf_model
+        self.model_selector_decay = model_selector_decay
+        self.nominal_attributes = nominal_attributes
+        self.max_depth = max_depth
+
+        if aggregation_method in self._VALID_AGGREGATION_METHOD:
+            self.aggregation_method = aggregation_method
+        else:
+            raise ValueError(f'Invalid aggregation_method: {aggregation_method}.\n'
+                             f'Valid values are: {self._VALID_AGGREGATION_METHOD}')
+
+    def predict_one(self, x: dict) -> base.typing.RegTarget:
+
+        if not self.models:
+            self._init_ensemble(features=list(x.keys()))
+            return 0.
+
+        y_pred = np.zeros(self.n_models)
+
+        if not self.disable_weighted_vote:
+            weights = np.zeros(self.n_models)
+            for idx, model in enumerate(self.models):
+                y_pred[idx] = model.predict_one(x)
+                weights[idx] = model.metric.get()
+
+                sum_weights = weights.sum()
+                if sum_weights != 0:
+                    # The higher the error, the worse is the tree
+                    weights = sum_weights - weights
+                    # Normalize weights to sum up to 1
+                    weights = weights / weights.sum()
+                    y_pred *= weights
+        else:
+            for idx, model in enumerate(self.models):
+                y_pred[idx] = model.predict_one(x)
+
+        if self.aggregation_method == self._MEAN:
+            y_pred = y_pred.mean()
+        else:
+            y_pred = np.median(y_pred)
```

Comment:
Nice solution!

Reply:
From this comment, I spotted a bug in the code. It is fixed now. Thanks... I guess :sweat_smile:

Reply:
Fix is in 53c75bb9b45bfa0539d0c4bcd86bd3707603c5d9